### ISSUE TITLE
SageMaker Example for HPO Analysis broken

### ISSUE THREAD

**Issue by marianokamp** [aws/amazon-sagemaker-examples#2911]
This [notebook](https://github.com/aws/amazon-sagemaker-examples/blob/master/hyperparameter_tuning/analyze_results/HPO_Analyze_TuningJob_Results.ipynb
) is our only example to analyze the results of tuning jobs and the other hyper parameter tuning notebooks are linking to it. 

It looks broken since this [PR Diff](https://app.reviewnb.com/awslabs/amazon-sagemaker-examples/pull/980/
).

**Descriptions**
The above linked Diff shows the exact changes. 

With the PR there is an expectation on the response structure of the [describe_hyper_parameter_tuning_job](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sagemaker.html#SageMaker.Client.describe_hyper_parameter_tuning_job) that is inconsisent with the linked documentation and the implementation, leading to a key error.

```
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-2-34bd1d6e8892> in <module>
     12 
     13 is_minimize = (
---> 14     tuning_job_result["TrainingJobDefinitions"][0]["TuningObjective"]["Type"] != "Maximize"
     15 )
     16 objective_name = tuning_job_result["TrainingJobDefinitions"][0]["TuningObjective"]["MetricName"]

KeyError: 'TrainingJobDefinitions'
```

 Old/New | Works for me? | Example
------- | ---- | ----
Old | Yes| [' HyperParameterTuningJobConfig']['HyperParameterTuningJobObjective'] 
New | No |  ['TrainingJobDefinitions'][0]['TuningObjective']

**To reproduce**
Create a new SM Classic Instance, checkout the samples, open the notebook, replace `YOUR-HYPERPARAMETER-TUNING-JOB-NAME` in the first code cell with a the job name of a hyper parameter tuninig job that ran in the same account. Upon execution of the 2nd code cell I run on a key error.

In the original [PR](https://github.com/aws/amazon-sagemaker-examples/pull/980) the author @guyernest stated:

> Issue #, if available:
The example notebook is not working any more due to the changes of the API
Description of changes:
Adding the name of the tuning job as the key for the results

Happy to get educated. As there seem to be multiple, overlapping issues. 
It would be good to better understand if there was an API change and what it was. The **current** documentation seems consistent with the **current** implementation, but not with this Notebook.

The error I am seeing, however, was **already** posted in the [comments](https://github.com/aws/amazon-sagemaker-examples/pull/980#pullrequestreview-591731651) of @aws-rbs on the merge in February. And he also mentioned that there was another issues as well. He states:

> But the existing version is broken too.

Apparently it was mixed up with the issue to not being able to test this notebook automatically in its current form, as it would require a HPO job. 

In short, when I roll back the change from above, all is great.

**Comment by jkroll-aws** [comment]
Notebook is fixed. Thanks for finding the specific commit that introduced the errors. Please reopen this issue if the updated version does not solve your problem.

One of the hyperparameter tuning jobs that I tested on was from September 2021 and it succeeds with the "old" response structure. I will follow up separately to make sure the boto3 documentation is consistent and is corrected as needed.
